source/slang/slang.cpp

```diff
@@ -205,11 +202,168 @@ CompileRequest::~CompileRequest()
     types = decltype(types)();
```

Comment:
I don't know the details here, because this is code that Yong wrote, but judging by the comment above those lines, I think this is a vestige of before we had the "IR-level type system."

With the old way of doing types in the IR, there were ways that the AST-level data structures could end up with pointers into IR structures, and if you deleted the IR first, those would be left as dangling pointers (I'm not sure where this led to crashes, but it did). By clearing out the AST structures that might point into the IR before the rest of the destructor runs, we avoided those issues.

None of this should be needed now that there is a clean separation between the AST and IR-level type systems. We could try out a cleanup change that just removes the lines here and see if it passes the tests.